pkg/odo/cli/url/create.go

```diff
@@ -118,7 +132,13 @@ func (o *URLCreateOptions) Complete(name string, cmd *cobra.Command, args []stri
 			return err
 		}
 
-		if len(args) == 0 {
+		if pushtarget.IsPushTargetDocker() {
```

Comment:
please also add a check that the exposed port flag is not used when the push target is not docker